tsl/src/nodes/decompress_chunk/decompress_chunk.c

```diff
@@ -377,13 +424,18 @@ cost_decompress_sorted_merge_append(PlannerInfo *root, DecompressChunkPath *dcpa
 	 * point. So, we use a quadratic cost model here to have higher costs than the normal
 	 * decompression when more than ~100 batches are used. We set
 	 * DECOMPRESS_CHUNK_HEAP_MERGE_CPU_TUPLE_COST to 0.8 to become most costly as soon as we have to
-	 * process more than 120 batches.
+	 * simultaneously hold more than 120 batches.
+	 *
+	 * Besides the component quadratic in the number of batches that are open at
+	 * the same time, we also have a component that is linear in the number of
+	 * total processed batches.
 	 *
 	 * Note: To behave similarly to the cost model of the regular decompression path, this cost
-	 * model does not consider the number of tuples.
+	 * model does not consider the number of decompressed tuples.
 	 */
 	dcpath->custom_path.path.total_cost =
-		sort_path.total_cost + pow(sort_path.rows, 2) * DECOMPRESS_CHUNK_HEAP_MERGE_CPU_TUPLE_COST;
```

Comment:
With this change, the costs for this path go up in the regression tests. Maybe we can simplify the cost model, remove the `pow()` function and do a check against work_memory to decide if we should use the optimization.

Reply:
Yeah, also the cost should probably be O(uncompressed_rows * log(open_batches)). What should we do with memory limit, just disable this optimization if it doesn't fit into work_mem?